it/test/uk/gov/hmrc/excisemovementcontrolsystemapi/repository/MovementRepositoryItSpec.scala

```diff
@@ -288,15 +296,91 @@ class MovementRepositoryItSpec extends PlaySpec
+        val expectedMovement3 =
+          Movement(Some("boxId"), "3", "consignorId1", Some("ern1"), Some("arc3"), lastUpdated = timestamp)
+        val expectedMovement4 = Movement(Some("boxId"), "4", "ern4", None, Some("arc4"), lastUpdated = timestamp)
+        insertMovement(expectedMovement1)
+        insertMovement(expectedMovement2)
+        insertMovement(expectedMovement3)
+        insertMovement(expectedMovement4)
+
+        val result =
+          repository.getMovementByERN(Seq("ern1"), MovementFilter.emptyFilter.copy(lrn = Some("1"))).futureValue
+
+        result mustBe Seq(
+          expectedMovement1
+        )
+
+      }
+
+      "arc from filter matches record from database" in {
+        val expectedMovement1 =
+          Movement(Some("boxId"), "1", "consignorId1", Some("ern1"), Some("arc1"), lastUpdated = timestamp)
+        val expectedMovement2 =
+          Movement(Some("boxId"), "2", "ern1", Some("ern2"), Some("arc2"), lastUpdated = timestamp)
+        val expectedMovement3 =
+          Movement(Some("boxId"), "3", "consignorId1", Some("ern1"), Some("arc3"), lastUpdated = timestamp)
+        val expectedMovement4 = Movement(Some("boxId"), "4", "ern4", None, Some("arc4"), lastUpdated = timestamp)
+        insertMovement(expectedMovement1)
+        insertMovement(expectedMovement2)
+        insertMovement(expectedMovement3)
+        insertMovement(expectedMovement4)
+
+        val result =
+          repository.getMovementByERN(Seq("ern1"), MovementFilter.emptyFilter.copy(arc = Some("arc1"))).futureValue
+
+        result mustBe Seq(
+          expectedMovement1
         )
+
+      }
+
+      "updatedSince from filter matches record from database" in {
+        val expectedMovement1 =
+          Movement(Some("boxId"), "1", "consignorId1", Some("ern1"), Some("arc1"), lastUpdated = timestamp)
+        val expectedMovement2 =
+          Movement(Some("boxId"), "2", "ern1", Some("ern2"), Some("arc2"), lastUpdated = timestamp)
+        val expectedMovement3 =
+          Movement(Some("boxId"), "3", "consignorId1", Some("ern1"), Some("arc3"), lastUpdated = timestamp)
+        val expectedMovement4 = Movement(Some("boxId"), "4", "ern4", None, Some("arc4"), lastUpdated = timestamp)
+        insertMovement(expectedMovement1)
+        insertMovement(expectedMovement2)
+        insertMovement(expectedMovement3)
+        insertMovement(expectedMovement4)
+
+        val filterTimestamp = LocalDateTime.of(2023, 1, 1, 12, 0).toInstant(ZoneOffset.MIN)
+
+        val result =
+          repository
+            .getMovementByERN(Seq("ern1"), MovementFilter.emptyFilter.copy(updatedSince = Some(filterTimestamp)))
+            .futureValue
+
+        result.sortBy(_.localReferenceNumber) mustBe Seq(
+          expectedMovement1,
+          expectedMovement2,
+          expectedMovement3
+        )
+
       }
     }
 
     "return an empty list" in {
```

Comment:
It would be good to see a test or a couple of tests to check that multiple filters applied has the "and" effect and the data is filtered to a specific movement (filter on ern and lrn or updatedSince and ern etc.)

Reply:
Done :)